### ISSUE TITLE
zypper_repository dont changes repo url

### ISSUE THREAD

**Issue by martingabelmann** [ansible/ansible-modules-extras#939]
##### Summary:

The module zypper_repository dont changes repositories if the given URL has changed.
##### Steps To Reproduce:

First I added a repo with

```
- name: add repository
  zypper_repository: name='OurRepo' repo='file://path/to/repo' state=present
```

which worked as expected. On the second run nothing changes.

Then we switched  to a http-repo, so I just changed the repo-url:

```
- name: add repository 
  zypper_repository: name='OurRepo' repo='http://rpm.exampe.org/repo/' state=present
```

but ansible didnt changed something until I removed the repo by hand and run it again.

Maybe there is something wrong with [L150](https://github.com/ansible/ansible-modules-extras/blob/devel/packaging/os/zypper_repository.py#L150) (I guess this part was for comparing if the repo-url/-name has changed) , or maybe i missunderstand what the 'expected behavior' should be.
##### Expected Results:

TASK: changed
on first run after edit.
##### Actual Results:

TASK: ok

**Comment by robynbergeron** [comment]
Hi @martingabelmann - thanks for submitting this bug report.

@matze -- pinging you to let you know about this bug report for zypper_repository.py.

**Comment by robynbergeron** [comment]
[module: packaging/os/zypper_repository.py]

**Comment by robinro** [comment]
2 scenarios that should work:

```
tasks:
- zypper_repository: alias=myoldalias url=mynewurl
- zypper_repository: alias=mynewalias url=mynewurl
```

and

```
tasks:
- zypper_repository: alias=mynewalias url=myoldurl
- zypper_repository: alias=mynewalias url=mynewurl
```

In both cases, the second task should be changed and the state afterwards should be `alias=mynewalias url=mynewurl`

**Comment by agx** [comment]
This is fixed by #1515. I'm happy to provide integration test once we've merged this to make sure we don't regress.
